### ISSUE TITLE
Connection lost toast persists forever for screen readers

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#12937]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The toast which reads "Connection lost. Reconnecting..." persists long after reconnection for screen reader users.  Although the visible toast may have been cleared, there is an off-screen [`<iron-a11y-announcer>`](https://github.com/PolymerElements/iron-a11y-announcer) element at the bottom of the DOM which continues to hold that text and announce it periodically according to the `aria-live` setting.  The only way to clear it is to refresh the page.

### Describe the behavior you expected

The `<iron-a11y-announcer>` should be removed after reconnection.

### Steps to reproduce the issue

Just examine the DOM after the toast and subsequent reconnection to see the element.

### What version of Home Assistant Core has the issue?

2022.6.5

### In which browser are you experiencing the issue with?

Firefox, Chrome, or Safari (latest stable)

### Which operating system are you using to run this browser?

Windows 10 and iOS 15

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Still in a shoe

**Comment by steverep** [comment]
This was fixed by the migration to `mwc-snackbar` in #19579.
